### ISSUE TITLE
Need a mechanism to match elements by a matching key rather than by element type

### ISSUE THREAD

**Issue by abhinav-k** [intuit/fuzzy-matcher#1]
Let us suppose we have documents with same elementTypes as follows
```
{
   name: NameElementType,
   spouseName: NameElementType
}
```
Need a mechanism to match these elements independently based on their key rather than element type.  In the above example, we want to match all documents based on name and spouseName independently.

**Comment by manishobhatia** [comment]
Added pull to resolve this issue
https://github.com/intuit/fuzzy-matcher/pull/7

Will be available in the next build

**Comment by manishobhatia** [comment]
fixed with 0.4.0 release
Element now has a "variance" that can be set to allow multiple ElementType's in a Document
